osu.Game/Rulesets/Objects/BezierConverter.cs

```diff
@@ -124,46 +124,46 @@ public static List<PathControlPoint> ConvertToModernBezier(IList<PathControlPoin
 
                 // The current vertex ends the segment
                 var segmentVertices = vertices.AsSpan().Slice(start, i - start + 1);
-                var segmentType = controlPoints[start].Type ?? PathType.Linear;
+                var segmentType = controlPoints[start].Type ?? PathType.LINEAR;
 
-                switch (segmentType)
+                switch (segmentType.Type)
                 {
-                    case PathType.Catmull:
+                    case SplineType.Catmull:
                         foreach (var segment in ConvertCatmullToBezierAnchors(segmentVertices))
                         {
                             for (int j = 0; j < segment.Length - 1; j++)
                             {
-                                result.Add(new PathControlPoint(segment[j], j == 0 ? PathType.Bezier : null));
+                                result.Add(new PathControlPoint(segment[j], j == 0 ? PathType.BEZIER : null));
                             }
                         }
 
                         break;
 
-                    case PathType.Linear:
+                    case SplineType.Linear:
                         foreach (var segment in ConvertLinearToBezierAnchors(segmentVertices))
                         {
                             for (int j = 0; j < segment.Length - 1; j++)
                             {
-                                result.Add(new PathControlPoint(segment[j], j == 0 ? PathType.Bezier : null));
+                                result.Add(new PathControlPoint(segment[j], j == 0 ? PathType.BEZIER : null));
                             }
                         }
 
                         break;
 
-                    case PathType.PerfectCurve:
+                    case SplineType.PerfectCurve:
                         var circleResult = ConvertCircleToBezierAnchors(segmentVertices);
 
                         for (int j = 0; j < circleResult.Length - 1; j++)
                         {
-                            result.Add(new PathControlPoint(circleResult[j], j == 0 ? PathType.Bezier : null));
+                            result.Add(new PathControlPoint(circleResult[j], j == 0 ? PathType.BEZIER : null));
                         }
 
                         break;
 
                     default:
                         for (int j = 0; j < segmentVertices.Length - 1; j++)
                         {
-                            result.Add(new PathControlPoint(segmentVertices[j], j == 0 ? PathType.Bezier : null));
+                            result.Add(new PathControlPoint(segmentVertices[j], j == 0 ? segmentType : null));
```

Comment:
This looks like it's going to preserve B-spline nodes of nonstandard degrees as they were. This is probably going to be an issue as the method being changed here is used to backport lazer sliders to stable, and stable cannot support the new `Bxx` curve specs.

I'm not yet sure how much of a problem this is going to be, and am yet to test, but yeah...

Reply:
Yeah this produces broken maps when exporting for compatibility to stable.

Thankfully, as far as I can tell, framework [already implements Boehm's algorithm](https://github.com/ppy/osu-framework/blob/2024a31b07e8896897312b79323b890f5b4e1583/osu.Framework/Utils/PathApproximator.cs#L84-L108) to remedy this, it just needs to be exposed for the game.